pkg/controller/shared_state.go

```diff
@@ -1135,20 +1135,37 @@ func (c *SharedState) cleanUpAzVolumeAttachmentByNode(ctx context.Context, azDri
 func (c *SharedState) cleanUpAzVolumeAttachments(ctx context.Context, attachments []azdiskv1beta2.AzVolumeAttachment, cleanUp attachmentCleanUpMode, caller operationRequester) error {
 	var err error
 
+	var markDetachRequest = func(attachment *azdiskv1beta2.AzVolumeAttachment, patchRequired *bool) {
+		attachment.Status.Annotations = azureutils.AddToMap(attachment.Status.Annotations, consts.VolumeDetachRequestAnnotation, string(caller))
+		*patchRequired = true
+	}
+
+	var markCleanUp = func(attachment *azdiskv1beta2.AzVolumeAttachment, patchRequired *bool) {
+		attachment.Status.Annotations = azureutils.AddToMap(attachment.Status.Annotations, consts.CleanUpAnnotation, string(caller))
+		*patchRequired = true
+	}
+
```

Comment:
Let's make `markDetachRequest` and `markCleanup` standalone functions.

Reply:
I believe we should create a standalone `addAnnotation` function instead.

Reply:
> np: I believe we should create a standalone addAnnotation function instead because in these methods we are not doing the update to complete the marking.
I agree that a standalone `addAnnotation` function would be useful. To your second point, I am not sure if I am onboard as our code already has the naming convention, as `updateState`, `updateStatusDetail`, `removeFinalizer`, etc., that modifies the object without making the Update API call.

Reply:
On second thought, I feel like `addAnnotation` would be a bit redundant as it doesn't add much extra to `AddToMap`'s functionality.